Sources/GRPCProtobufParser/ProtobufToCodeGenParser.swift

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import SwiftProtobuf
+import SwiftProtobufPluginLibrary
+
+import struct GRPCCodeGen.CodeGenerationRequest
+
+/// Parses a ``FileDescriptor`` object into a ``CodeGenerationRequest`` object.
+public struct ProtobufToCodeGenParser {
```

Comment:
Given we now have this extra module we can put more of the logic in it when constructing `protoc-gen-grpc-swift`, and have `protoc-gen-grpc-swift` basically do nothing apart from parsing the options and calling the generator we provide in this module. The upshot of this is that I don't think this type needs to be `public`.